pkg/controller/certificates/certificate_controller.go

```diff
@@ -191,16 +193,17 @@ func (cc *CertificateController) syncFunc(key string) error {
 	return cc.handler(csr)
 }
 
-// IgnorableError returns an error that we shouldn't handle (i.e. log) because
+// NewIgnorableError returns an error that we shouldn't handle (i.e. log) because
 // it's spammy and usually user error. Instead we will log these errors at a
 // higher log level. We still need to throw these errors to signal that the
 // sync should be retried.
-func IgnorableError(s string, args ...interface{}) ignorableError {
-	return ignorableError(fmt.Sprintf(s, args...))
+func NewIgnorableError(s string, args ...interface{}) IgnorableError {
+	return IgnorableError(fmt.Sprintf(s, args...))
 }
 
-type ignorableError string
+// IgnorableError is an error that can be ignored (e.g. user error)
+type IgnorableError string
```

Comment:
wouldn't it would be possible to keep the type private?
its seems to be only used in this package.

Reply:
If left unexported, golint says `exported func NewIgnorableError returns unexported type certificates.ignorableError, which can be annoying to use`. NewIgnorable error is used in the approver package (vs. certificates).

Reply:
NewIgnorableError should return just `error`